CONTRIBUTING.md

```diff
@@ -550,6 +550,239 @@ cat generated/watcher/watcher.log  # See k_ids field for each core in the last d
   ensure that your commit message follows our required format as outlined above
   in this section. Failure to do so is a violation of our standards.
 
+### Git commit runbook
+
+The tt-metal repository has a single main branch, with pull requests being
+merged in via rebase.
+
+For a pull request, you will need to deal with a local branch and an origin
+branch.
+
+The local branch should be on your VM, and you should interact with it through
+the command line or Visual Studio. The following uses the command line.
+
+The origin branch is on Github and you should use the Github UI to interact
+with the branch.
+
+#### Creating the initial Pull Request (PR)
+
+You need to have an issue. Either someone creates one and assigns it to you, or
+you need to create an issue. For example issue 123.
+
+#### Creating a branch on a VM
+
+Include the user, the issue number, and optionally a description of the change.
+/ and - are used as separators between user and issue number. And - and _
+between issue number and description. E.g.
+
+```
+git checkout -b user-123
+git checkout -b user/123
+git checkout -b user-123_rename_method_x
+git checkout -b user/123-add-x-unit-test
+```
+
+#### Saving your changes
+
+Edit the files that you want, making sure relevant unit tests pass. Then add
+them in. E.g.
+
+```
+git add abc.py
+git add -A
+```
+
+You can restore files if you need to get the original. E.g.
+
+```
+git restore abc.py
+git restore '*'
+git restore --staged abc.py # if the file was already added
+```
+
+Once you are satisfied that everything works, create a commit. The message
+needs to specify the issue number, with a pound sign and a colon, and the
+description should describe the change itself, and cannot be generic and refer
+to something else, such as saying that you are addressing reviewer feedback.
+
+```
+git commit -m"#123: rename method x"
+```
+
+Note: each commit should work. That is, everything compiles properly on the
+architecture used by your VM, you can run relevant code on the card, and
+relevant unit tests pass.
```

Comment:
Each commit on main branch is supposed to work.
Same apply for a feature branch where multiple engineers collaborate.
And I don't care what happens in your branch unless as soon as you respect the above 2 rules (and e.g. squash merge or rebase final result into working commit/s)

Overall one can't guarantee that a commit works without running CI. 
Forcing each commit on a feature branch to always work means a person is forced to spend time rebasing/changing local history instead of iterating to get things done. 

More than that, rebase requires a force push, which makes it very easy to accidentally ruin your work (specially in combination with things like `git add -A`. In most cases squash merge is more than sufficient, allows to simplify and de-risk most common workflows.

Reply:
The rest of the paragraph had details about what I meant, which isn't as strict as having CI pass. Regardless, updated the text.

I found a lot of times I need to rebase to fix conflicts, and it's good to have details about what to do.